xonsh/built_ins.py

```diff
@@ -720,6 +724,7 @@ def load_builtins(execer=None, config=None, login=False, ctx=None):
     builtins.__xonsh_execer__ = execer
     builtins.__xonsh_commands_cache__ = CommandsCache()
     builtins.__xonsh_all_jobs__ = {}
+    builtins.__xonsh_enabled_builtin_commands__ = set(all_builtin_commands.keys())
```

Comment:
Yeah, we are definitely abusing terms all over the place here :)  I kind of just want a single point of entry, and aliases is bad for that.  `commands` is what I thought too, but it isn't as snappy as I would like.  It is worth thinking about.

> I don't agree with this. Since aliases are loaded first, it is still just as easy to override the behavior of a built-in (just add an alias, as before!).

Yeah, I think my objection is on the other side of this. Say one wants to remove a command that is masking an installed binary (eg cat), do you check these builtins or aliases?  I agree that the story on adding masks and mapping is fine as it is.